File: scanner-app/lib/models/event.dart
class Event {
  Event({
    required this.id,
    required this.name,
    required this.slug,
    required this.publicKey,
    this.lastSynced,
  });

  final String id;
  final String name;
  final String slug;
  final String publicKey;
  final String? lastSynced;

  Map<String, dynamic> toMap() {
    return {
      "id": id,
      "name": name,
      "slug": slug,
      "public_key": publicKey,
      "last_synced": lastSynced,
    };
  }

  factory Event.fromMap(Map<String, dynamic> map) {
    return Event(
      id: map["id"] as String,
      name: map["name"] as String,
      slug: map["slug"] as String,
      publicKey: map["public_key"] as String,
      lastSynced: map["last_synced"] as String?,
    );
  }
}

File: backend/src/db/models/Event.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IEvent extends Document {
  slug: string;
  name: string;
  description: string;
  date: Date;
  location: string;
  status: 'DRAFT' | 'ACTIVE' | 'COMPLETED';
  sheetId: string;
  qrPublicKey: string;
  qrPrivateKey: string;
  createdAt: Date;
  updatedAt: Date;
}

const EventSchema = new Schema<IEvent>(
  {
    slug: {
      type: String,
      required: true,
      unique: true,
      lowercase: true,
      trim: true,
    },
    name: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      default: '',
    },
    date: {
      type: Date,
      required: true,
    },
    location: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: ['DRAFT', 'ACTIVE', 'COMPLETED'],
      default: 'DRAFT',
    },
    sheetId: {
      type: String,
      default: '',
    },
    qrPublicKey: {
      type: String,
      required: true,
    },
    qrPrivateKey: {
      type: String,
      required: true,
      select: false, // Don't include in queries by default (security)
    },
  },
  {
    timestamps: true,
  }
);

// slug index is already created by unique: true
EventSchema.index({ status: 1 });

export const Event = mongoose.model<IEvent>('Event', EventSchema);
